src/Compilers/Core/CodeAnalysisTest/Collections/TopologicalSortTests.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.Collections
+{
+    public class TopologicalSortTests
+    {
+        [Fact]
+        public void Test01()
+        {
+            int[][] successors = new int[][]
+            {
+                /* 0 */ new int[] { }, // 0 has no successors
+                /* 1 */ new int[] { },
+                /* 2 */ new int[] { 3 },
+                /* 3 */ new int[] { 1 },
+                /* 4 */ new int[] { 0, 1 },
+                /* 5 */ new int[] { 0, 2 },
+            };
+
+            var sorted = TopologicalSort.IterativeSort<int>(new[] { 4, 5 }, x => successors[x]);
+            AssertTopologicallySorted(sorted, successors, "Test01");
```

Comment:
Nit: Passing the name of the test is odd. Isn't the stacktrace on assertion sufficient?

Reply:
Yes, for this test it is. For others it isn't as there is a seed parameter.